api/pod-handlers.go

```diff
@@ -263,34 +269,37 @@ func getDescribePodResponse(session *models.Principal, params operator_api.Descr
 
 	}
 	for i := range pod.Spec.Containers {
+		container := pod.Spec.Containers[i]
 		retval.Containers[i] = &models.Container{
-			Name:      pod.Spec.Containers[i].Name,
-			Image:     pod.Spec.Containers[i].Image,
-			Ports:     describeContainerPorts(pod.Spec.Containers[i].Ports),
-			HostPorts: describeContainerHostPorts(pod.Spec.Containers[i].Ports),
-			Args:      pod.Spec.Containers[i].Args,
+			Name:      container.Name,
+			Image:     container.Image,
+			Ports:     describeContainerPorts(container.Ports),
+			HostPorts: describeContainerHostPorts(container.Ports),
+			Args:      container.Args,
 		}
-		if slices.Contains(statusKeys, pod.Spec.Containers[i].Name) {
-			retval.Containers[i].ContainerID = statusMap[pod.Spec.Containers[i].Name].ContainerID
-			retval.Containers[i].ImageID = statusMap[pod.Spec.Containers[i].Name].ImageID
-			retval.Containers[i].Ready = statusMap[pod.Spec.Containers[i].Name].Ready
-			retval.Containers[i].RestartCount = int64(statusMap[pod.Spec.Containers[i].Name].RestartCount)
-			retval.Containers[i].State, retval.Containers[i].LastState = describeStatus(statusMap[pod.Spec.Containers[i].Name])
+		if slices.Contains(statusKeys, container.Name) {
+			containerStatus := statusMap[container.Name]
+			retval.Containers[i].ContainerID = containerStatus.ContainerID
+			retval.Containers[i].ImageID = containerStatus.ImageID
+			retval.Containers[i].Ready = containerStatus.Ready
+			retval.Containers[i].RestartCount = int64(containerStatus.RestartCount)
+			retval.Containers[i].State = describeContainerState(containerStatus.State)
+			retval.Containers[i].LastState = describeContainerState(containerStatus.LastTerminationState)
 		}
-		retval.Containers[i].EnvironmentVariables = make([]*models.EnvironmentVariable, len(pod.Spec.Containers[i].Env))
-		for j := range pod.Spec.Containers[i].Env {
+		retval.Containers[i].EnvironmentVariables = make([]*models.EnvironmentVariable, len(container.Env))
+		for j := range container.Env {
 			retval.Containers[i].EnvironmentVariables[j] = &models.EnvironmentVariable{
-				Key:   pod.Spec.Containers[i].Env[j].Name,
-				Value: pod.Spec.Containers[i].Env[j].Value,
+				Key:   container.Env[j].Name,
+				Value: container.Env[j].Value,
 			}
 		}
-		retval.Containers[i].Mounts = make([]*models.Mount, len(pod.Spec.Containers[i].VolumeMounts))
-		for j := range pod.Spec.Containers[i].VolumeMounts {
+		retval.Containers[i].Mounts = make([]*models.Mount, len(container.VolumeMounts))
+		for j := range container.VolumeMounts {
```

Comment:
Could be changed to `for _, volume := range container.VolumeMounts`

Reply:
done